### ISSUE TITLE
Responsive sidenav doesn't update focus trap

### ISSUE THREAD

**Issue by phyrephox** [angular/components#16601]
#### Reproduction

Use StackBlitz to reproduce your issue: https://stackblitz.com/edit/angular-lkt2qw

Steps to reproduce:
1. Open the sidenav with the hamburger menu
2. Shrink the stackblitz preview window until it switches to 'over' mode
3. Tab through the sidenav
 
 
#### Expected Behavior

As the sidenav is now in over mode, the focus trap should be enabled.

#### Actual Behavior

The focus trap is not enabled so the user can tab past the sidenav into the main content.
This also happens in reverse, when opening the sidenav in over mode and switching to side mode. The focus trap remains enabled even though the user should be able to tab away from the sidenav.

#### Environment

- Angular: 8.x
- CDK/Material: 8.x
- Browser(s): Tested in chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Ubuntu, ChromeOS

I think the issue is [this line](https://github.com/angular/components/blob/8ecfc72269a58e0e1cb4a7c40b030ebf26de9310/src/material/sidenav/drawer.ts#L249), where the focus trap is only updated on open/close, rather than mode change as well.
